unit-tests/dds/test-options.py

```diff
@@ -0,0 +1,103 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#test:donotrun:!dds
+
+import pyrealdds as dds
+from rspy import log, test
+
+
+dds.debug( True, 'C  ' )
+log.nested = 'C  '
+
+
+participant = dds.participant()
+participant.init( 123, "test-options" )
+
+info = dds.device_info()
+info.name = "Test Device"
+info.topic_root = "realdds/device/topic-root"
+
+
+import os.path
+cwd = os.path.dirname(os.path.realpath(__file__))
+remote_script = os.path.join( cwd, 'options-server.py' )
+with test.remote( remote_script, nested_indent="  S" ) as remote:
+    remote.wait_until_ready()
+    #
+    #############################################################################################
+    #
+    test.start( "Test no options..." )
+    try:
+        remote.run( 'test_no_options()', timeout=5 )
+        device = dds.device( participant, participant.create_guid(), info )
+        device.run()  # If no device is available in 30 seconds, this will throw
+        test.check( device.is_running() )
+        
+        options = device.options();
+        for option in options:
+            test.unreachable() # Test no device option
+        
+        test.check_equal( device.n_streams(), 1 )
+        for stream in device.streams():
+            options = stream.options();
+            for option in options:
+                test.unreachable() # Test no stream option
+                
+        remote.run( 'close_server()', timeout=5 )
+    except:
+        test.unexpected_exception()
+    device = None
+    test.finish()
+    #
+    #############################################################################################
+    #
+    test.start( "Test device options discovery..." )
+    try:
+        remote.run( 'test_device_options_discovery()', timeout=5 )
+        device = dds.device( participant, participant.create_guid(), info )
+        device.run()  # If no device is available in 30 seconds, this will throw
+        test.check( device.is_running() )
+        
+        options = device.options();
+        test.check_equal( len( options ), 3 )
```

Comment:
I think that it will be clearer if options-server will be more generic and will get the values here, is this possible?

Here we compare the values to values that were introduces somewhere else

Reply:
Done